src/proxy/metrics.rs

```diff
@@ -108,6 +109,8 @@ pub struct ConnectionClose<'a>(&'a ConnectionOpen);
 
 pub struct BytesTransferred<'a>(&'a ConnectionOpen);
 
+pub struct ConnectionDenied<'a>(&'a ConnectionOpen);
```

Comment:
We report denied in Envoy as well -- but as a response flag. Should we do the same here?

Reply:
Could you explain a little bit more what you mean by that. In the context of a connection, those are mostly used for debugging/logging purposes. There's no real way to return them to the user.

Reply:
I mean we already have a way to report "connection denied by RBAC" in the existing metrics scheme, as we have a response_flag label that indicates why it was closed. Seems like if that works for envoy it should work for us, too.

Reply:
Ok, I think I see what you're saying now. Currently that's a `CommonTrafficLabel`, but I'm not sure `connection denied by RBAC` really makes sense on any of the existing metrics` ConnectionOpen`. Are you saying put the reason on `connection_closed` instead?

I do think there could be value in having a separate metric specifically for RBAC, especially to implement `AUDIT` in the future. Having context data as multiple extra labels could get confusing.

Reply:
I don't know if we put it on open or closed in Istio envoy TCP, we can check. For HTTP its more clear since there is just "Requests" and we put code=403,flag=<something>.

Reply:
So I looked at the code, and the TCP metrics are all a bit async because the reportHelper is called on a timer.
However, it's called with endStream [here](https://github.com/istio/proxy/blob/ec5563aa0ff7333af61d51b4d5837f218c19b70b/source/extensions/filters/http/istio_stats/istio_stats.cc#L790C5-L790C17), and then the response code is added to the `tags_` [a few lines down](https://github.com/istio/proxy/blob/ec5563aa0ff7333af61d51b4d5837f218c19b70b/source/extensions/filters/http/istio_stats/istio_stats.cc#L798). Basically it doesn't seem like it's meant to show up for TCP metrics. Which in my opinion makes sense, it's much more of an HTTP concept. 

More reason I suppose to do something like you [mentioned above](https://github.com/istio/ztunnel/pull/664#discussion_r1304515927)

Reply:
I agree we should start with the same format as envoy for these metrics; users should not have to scrape two different formats

Reply:
The problem with doing the same format as `envoy` for this particular thing is that the denies are actually captured and returned by the RBAC filter. So the stats will be `RBAC` stats, not `istio` stats. You can make some inference based on the response_code, but that could also come from upstream so it's not as accurate.

Reply:
I think the confusion here is that I was wrong :slightly_smiling_face:.

In metrics, we have "response flag". In ztunnel it is always `-`.

In logs there is also `RESPONSE_CODE_DETAILS`. _That_ is where rbac is reported (also, envoy docs lie and say its only HTTP -- its not true).

To me an ideal outcome would be to define a new response flag (perhaps `RBACDenied (RD)`) in envoy. Implement it for envoy RBAC, which is then reporting in metrics and logs. Report the same flag in ztunnel.

WDYT?

Technically we can have ztunnel specific response flags as well but not ideal